src/objects/texts/zcl_abapgit_longtexts.clas.abap

```diff
@@ -162,15 +162,24 @@ CLASS zcl_abapgit_longtexts IMPLEMENTATION.
   METHOD zif_abapgit_longtexts~deserialize.
 
     DATA: lt_longtexts    TYPE ty_longtexts,
+          lv_longtext_id  TYPE dokil-id,
+          lv_object_name  TYPE dokil-object,
+          lt_dokil        TYPE zif_abapgit_definitions=>ty_dokil_tt,
           lv_no_main_lang TYPE dokil-masterlang.
-    FIELD-SYMBOLS: <ls_longtext> TYPE ty_longtext.
+
+    FIELD-SYMBOLS: <ls_longtext> TYPE ty_longtext,
+                   <ls_dokil>    TYPE dokil.
 
     ii_xml->read(
       EXPORTING
         iv_name = iv_longtext_name
       CHANGING
         cg_data = lt_longtexts ).
 
+    IF lt_longtexts IS INITIAL.
+      RETURN.
```

Comment:
If there are no longtexts at all in the xml then the deletion logic below will not be executed?

Reply:
yes. needs some refactoring of the interface to pass in the id/name

Reply:
done